src/Quaternion.cs

```diff
@@ -132,26 +132,39 @@ public static Quaternion LookAt(Vector3 position, Vector3 targetPosition, Vector
             var projectedTarget = Plane.ProjectPointOntoPlane(plane, targetPosition);
             var projectedDirection = (projectedTarget - position).Normalize();
 
-            var q1 = CreateRotationFromAToB(forward, projectedDirection);
-            var q2 = CreateRotationFromAToB(projectedDirection, (targetPosition - position).Normalize());
+            var q1 = CreateRotationFromAToB(forward, projectedDirection, up);
+            var q2 = CreateRotationFromAToB(projectedDirection, (targetPosition - position).Normalize(), up);
 
             return q2 * q1;
         }
 
         /// <summary>
-        /// Creates a new Quaternion rotating vector 'fromA' to 'toB'
+        /// Creates a new Quaternion rotating vector 'fromA' to 'toB'.<br/>
+        /// Precondition: fromA and toB are normalized.
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static Quaternion CreateRotationFromAToB(Vector3 fromA, Vector3 toB)
+        public static Quaternion CreateRotationFromAToB(Vector3 fromA, Vector3 toB, Vector3? up = null)
         {
             var axis = fromA.Cross(toB);
             var lengthSquared = axis.LengthSquared();
             if (lengthSquared > 0.0f)
             {
                 return CreateFromAxisAngle(axis / (float)Math.Sqrt(lengthSquared), (float)Math.Acos(fromA.Dot(toB)));
             }
-
-            return Identity;
+            else
+            {
+                // The vectors are parallel to each other
+                if ((fromA + toB).AlmostZero())
```

Comment:
I believe this check needs to happen earlier. if this is true in the if statement above, the second argument in CreatFromAxisAngle - float)Math.Acos(fromA.Dot(toB)) - will be NaN